src/conversions.jl

```diff
@@ -705,6 +712,59 @@ function convert_attribute(ls::Symbol, ::key"linestyle")
     end
 end
 
+function line_diff_pattern(ls_str::AbstractString, gaps = :normal)
+    dot = 1
+    dash = 3
+    check_line_pattern(ls_str)
+    
+    dot_gap, dash_gap = convert_gaps(gaps)
+    
+    pattern = Float64[]
+    for i in 1:length(ls_str)
+        curr_char = ls_str[i]
+        next_char = i == lastindex(ls_str) ? ls_str[begin] : ls_str[i+1]
+        # push dash or dot
+        if curr_char == '-'
+            push!(pattern, dash)
+        else
+            push!(pattern, dot)
+        end
+        # push the gap (use dot_gap only between two dots)
+        if (curr_char == '.') && (next_char == '.')
+            push!(pattern, dot_gap)
+        else
+            push!(pattern, dash_gap)
+        end
+    end
+    pattern
+end
+
+"Checks if the linestyle format provided as a string contains only dashes and dots"
+function check_line_pattern(ls_str)
+    isnothing(match(r"^[.-]+$", ls_str)) &&
+        throw(ArgumentError("If you provide a string as linestyle, it must only consist of dashes (-) and dots (.)"))
+end
```

Comment:
ah yeah now you have to return nothing here to pass the tests, otherwise a Match will be returned

Reply:
right, I should probably run tests locally before I push. -_-

Reply:
... but I didn't.